server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java

```diff
@@ -163,6 +167,13 @@ public Optional<SyntaxError> registerParagraphNameNode(ParagraphNameNode node) {
    * @return syntax error if the code block duplicates
    */
   public Optional<SyntaxError> registerSectionNameNode(SectionNameNode node) {
+    if (sectionMap.containsKey(node.getName())) {
```

Comment:
Maybe here better to create a new method `boolean containSection(String sectionName)` and move this logic to the place where you call this `registerSectionNameNode` method.

Reply:
Maybe... but not much. fixed.